### PR TITLE
Improve pylint quality across code base

### PR DESCRIPTION
**Pull request by dwierichs**
This PR realizes easy-to-implement improvements regarding linting across the codebase.

### CODE REVIEW

File: pennylane/ops/qubit/matrix_ops.py
Comment (dwierichs) [reply pennylane/ops/qubit/matrix_ops.py:246]: Not mine, pylint's! :D

File: pennylane/transforms/optimization/cancel_inverses.py
Comment (dwierichs) [reply pennylane/transforms/optimization/cancel_inverses.py:117]: Yes, this change reduces the maximal block indentation level from 6 to 5, i.e. to the "allowed" maximum.
Based on the code duplication, it feels like this can be simplified/restructured further. Essentially there only ever happen two things in the leafs of the nested if-else clauses.

File: pennylane/qaoa/cycle.py
Comment (dwierichs) [reply pennylane/qaoa/cycle.py:440]: Oh dear, I did not see this `disable` comment. It actually does not affect pylint, and it complained about the line that I changed... I'll move the comment to the same line to "activate" it, and revert my change.

Comment (josh146) [reply pennylane/qaoa/cycle.py:440]: I went through the git blame, and it seems like this is the history here:

1. the `type(op1) == type(op2)` comparison was added in a PR. Codefactor allowed it due to the zero-sum behaviour.

2. A later PR was blocked by Codefactor, so someone added `# pylint: disable=unidiomatic-typecheck`.

So @dwierichs your 'fix' here (alongside removing the pylint disable comment) could be valid. Let's just wait if @trbromley or @anthayes92 have anything to add before we merge though.

File: pennylane/tracker.py
Comment (josh146) [inline pennylane/tracker.py:103]: pylint caught this from inside a docstring??

**Review by josh146 — APPROVED** [review]
Thanks for this @dwierichs! This looks good from my end, but might be good just to hear back from the people that I've tagged for confirmation before merging

File: pennylane/tracker.py
Comment (dwierichs) [reply pennylane/tracker.py:103]: No, it caught one version of this usage in the code and I noticed (via search-in-code) that there is another in the docstring :)
I thought it might be nice to sync docstring and codebase styles :)

File: pennylane/transforms/optimization/cancel_inverses.py
Comment (dwierichs) [reply pennylane/transforms/optimization/cancel_inverses.py:118]: By moving the final `list_copy.pop(0)` further up, I made use of `continue` statements a bit more and (imho) improved the readability a bit :)

File: pennylane/hf/matrices.py
Comment (soranjh) [inline pennylane/hf/matrices.py:93]: Just double checking: it contains the condition `if i < j`? If this condition is not there, the tests will still pass but we will be computing repetitive terms. So the efficiency will be affected.

File: pennylane/hf/matrices.py
Comment (soranjh) [inline pennylane/hf/matrices.py:141]: Just double checking: it contains the condition `if i < j`? If this condition is not there, the tests will still pass but we will be computing repetitive terms. So the efficiency will be affected. 

Why `combinations_with_replacement` is used here but not in the previous function?

File: pennylane/hf/matrices.py
Comment (soranjh) [inline pennylane/hf/matrices.py:191]: Just double checking: it contains the condition `if i < j`? If this condition is not there, the tests will still pass but we will be computing repetitive terms. So the efficiency will be affected.

File: pennylane/hf/matrices.py
Comment (soranjh) [inline pennylane/hf/matrices.py:287]: This looks better!

File: pennylane/hf/integrals.py
Comment (soranjh) [reply pennylane/hf/integrals.py:609]: Seems good to me.

File: pennylane/hf/integrals.py
Comment (soranjh) [reply pennylane/hf/integrals.py:738]: Seems good.

**Review by josh146 — APPROVED** [review]

**Pull request merged**
